plugins/pulp_rpm/plugins/migrations/0025_export_relative_url.py

```diff
@@ -0,0 +1,64 @@
+"""
+Migration to normalize relative_url, and populate relative_url on
+export distributors
+"""
+from pulp.server.db import connection
+
+
+def _find_yum_distributor(distributors):
+    """
+    Find this repo's 'yum_distributor' to pull the relative_url from.
+
+    While technically you can have more than one yum_distributor...
+    functionality really only works for a single distributor at a time.
+    This distributor's ID is 'yum_distributor'
+
+    :param distributors: List of distributors for repo.
+    """
+    for distributor in distributors:
+        if distributor['distributor_id'] == 'yum_distributor':
+            if 'config' in distributor and 'relative_url' in distributor['config']:
+                return distributor
+    return None
+
+
+def _clean_distributors_relative_url(repo_distributors, distributors):
+    for distributor in distributors:
+        if 'config' in distributor and 'relative_url' in distributor['config']:
+            relative_url = distributor['config']['relative_url'].lstrip('/')
+            distributor['config']['relative_url'] = relative_url
+            repo_distributors.update({'_id': distributor['_id']},
+                                     {'$set': {'config': distributor['config']}})
+
+
+def migrate(*args, **kwargs):
+    """
+    Perform the migration as described in this module's docblock.
+
+    :param args:   unused
+    :type  args:   list
+    :param kwargs: unused
+    :type  kwargs: dict
+    """
+    db = connection.get_database()
+    repos = db['repos']
+    repo_distributors = db['repo_distributors']
+    repo_objects = repos.find({'notes': {'_repo-type': 'rpm-repo'}})
+    for repo_object in repo_objects:
+        distributors = list(repo_distributors.find({'repo_id': repo_object['repo_id']}))
```

Comment:
Not casting to a list caused issues iterating over the object multiple times.  Since the result of find is a cursor, each time I got the next element from find it advanced the cursor (in the nested looping).  Essentially causing the iteration to end prematurely.  Casting to a list allows a new iterator for each for x in distributor case for the full list.